python/pyarrow/fs.py

```diff
@@ -356,7 +356,18 @@ def get_file_info_selector(self, selector):
             selector.base_dir, maxdepth=maxdepth, withdirs=True, detail=True
         )
         for path, info in selected_files.items():
-            infos.append(self._create_file_info(path, info))
+            start = start_p = 0
+            end = len(selector.base_dir)
+            if path.startswith("/"):
+                start_p += 1
+            if selector.base_dir.startswith("/"):
+                start += 1
+            if selector.base_dir.endswith("/"):
+                end -= 1
```

Comment:
Would it be best to also add `end_p` in case fsspec changes in the future or some implementation differs slightly in the future?

Reply:
Why not use `str.strip` or `str.lstrip` here?

Reply:
Ah, good call. I forgot `strip()` takes from both ends. I was going to suggest lstrip + rstrip, but that was two string copies instead of one copy with an array slice.

Reply:
👍 Will change.